### ISSUE TITLE
iOS 12.2 real device web tests fail

### ISSUE THREAD

**Issue by imurchie** [appium/appium#12398]
We cannot currently support web/hybrid on iOS 12.2.

I am working on figuring out the communication problems, and then will probably need to update `appium-remote-debugger` as we did for sims.

**Comment by imurchie** [comment]
`brew install ios-webkit-debug-proxy --HEAD` should now include https://github.com/google/ios-webkit-debug-proxy/pull/300 which will help with communication.

**Comment by KazuCocoa** [comment]
I found below errors like below log on real device x iOS 12.2. (With current Appium master and the head of ios-webkit-debug-proxy).
Does this issue include this error as well?

```
[debug] [iOS] Picking webview 'WEBVIEW_1'
[debug] [iOS] Attempting to set context to 'WEBVIEW_1'
[debug] [RemoteDebugger] WebKit debugger web socket connected to url: ws://localhost:27753/devtools/page/1
[debug] [RemoteDebugger] Starting to listen for JavaScript console
[debug] [RemoteDebugger] Sending WebKit data: {"method":"Console.enable","params":{"objectGroup":"console","includeCommandLineAPI":true,"doNotPauseOnExceptionsAndMuteConsole":true}}
[debug] [RemoteDebugger] Webkit response timeout: 5000
[debug] [RemoteDebugger] Received WebKit data: '{"method":"Target.targetCreated","params":{"targetInfo":{"targetId":"page-1","type":"page"}}}'
[debug] [RemoteDebugger] Found method 'Target.targetCreated'
[debug] [RemoteDebugger] There is no handler scheduled for method 'Target.targetCreated' in the recent message
[debug] [RemoteDebugger] Received WebKit data: '{"error":{"code":-32601,"message":"'Console' domain was not found","data":[{"code":-32601,"message":"'Console' domain was not found"}]},"id":1}'
[debug] [RemoteDebugger] Found method 'Console.enable' for '1' message
[XCUITest] [object Object]
```

(https://gist.github.com/KazuCocoa/5aa09c3e7af485b8007c5e35c91d299b is the full log)

`Page`, `Network` also happened.

I dug https://opensource.apple.com/tarballs/WebInspectorUI/ a couple of hours. Maybe, such error happens commands in https://opensource.apple.com/source/WebInspectorUI/WebInspectorUI-7600.1.17/UserInterface/Protocol/Legacy/ ...
I commented  `Console.enable` etc out once. Then, _create session_ succeeded, but open an arbitrary page did not work since `Page.navigate` also raised the same error...

Safari's inspector worked on macOS 10.14.4 x iOS 12.2. So, maybe we can find another way...

**Comment by imurchie** [comment]
@KazuCocoa See https://github.com/appium/appium-remote-debugger/pull/122

**Comment by milenyordanov** [comment]
I had a working setup with appium 1.10 and iOS 12.1 real device running web tests on mobile Safari.
I upgraded the device to iOS 12.2, XCode to 10.2, appium to 1.12.1. But now I cannot get the tests to run. Is this issue about a scenario like mine?
The last lines from my log are:
```
appium stderr:
2019-04-03 16:16:06:861 - [Xcode]     t =     0.74s Open com.apple.mobilesafari

appium stderr:
2019-04-03 16:16:06:861 - [Xcode] 

appium stderr:
2019-04-03 16:16:06:865 - [Xcode]     t =     0.74s     Launch com.apple.mobilesafari

appium stderr:
2019-04-03 16:16:06:865 - [Xcode] 

appium stderr:
2019-04-03 16:16:07:141 - [Xcode]     t =     1.02s         Wait for com.apple.mobilesafari to idle

appium stderr:
2019-04-03 16:16:07:141 - [Xcode] 

appium stderr:
2019-04-03 16:16:08:463 - [Xcode]     t =     2.34s Find the Application 'com.apple.mobilesafari'

appium stderr:
2019-04-03 16:16:08:463 - [Xcode] 

appium stderr:
2019-04-03 16:16:08:478 - [Xcode]     t =     2.35s     Snapshot accessibility hierarchy for app with pid 465

appium stderr:
2019-04-03 16:16:08:479 - [Xcode] 

appium stderr:
2019-04-03 16:16:09:402 - [Xcode]     t =     3.28s Activate Siri with voice recognition text: Open {https://google.com}

appium stderr:
2019-04-03 16:16:09:402 - [Xcode] 

appium stderr:
2019-04-03 16:16:15:525 - [Xcode]     t =     9.40s     Wait for accessibility to load

appium stderr:
2019-04-03 16:16:15:525 - [Xcode] 

appium stderr:
2019-04-03 16:17:09:397 - [XCUITest] UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: ESOCKETTIMEDOUT
```

**Comment by milenyordanov** [comment]
Also the Console on iOS device shows this error:
```
fault	15:31:55.451454 +0300	MobileSafari	The -u command line option has been removed (rdar://42146540), please file a radar with your use case if you rely on this
```

**Comment by mahendramohan** [comment]
Hi,
I am getting the below errors. Safari browser is launching but somehow, the URL are not launching

`[WD Proxy] Got an unexpected response: {"errno":"ECONNRESET","code":"ECONNRESET","syscall":"read"}`

and also,


```
[WD Proxy] Proxying [POST /url] to [POST http://localhost:8100/session/CEEF7269-817A-4D91-96AC-963E53E93EE3/url] with body: {"url":"<URL>"}
[Xcode]     t =    62.04s Activate Siri with voice recognition text: Open {<URL>}
```

I would like to know whether the issue is resolved and if there is any alternative approach for this issue.

Thanks in advance

**Comment by milen-yordanov** [comment]
@mahendramohan Try to enable Siri. Siri opens the browser with the URL.

**Comment by mahendramohan** [comment]
> @mahendramohan Try to enable Siri. Siri opens the browser with the URL.

It is already enabled in the Xcode capabilities using developer profile.

**Comment by imurchie** [comment]
Yes. This is fixed in a PR, which we are close to getting merged.

https://github.com/appium/appium-xcuitest-driver/pull/918

**Comment by mahendramohan** [comment]
@imurchie From where we can get the fixed PR??

**Comment by dpgraham** [comment]
Next release will be out in April with this fix.... this can also be tested in beta `npm uninstall -g appium && npm install -g appium@beta`.

**Comment by simplezhang1990** [comment]
@dpgraham  when this issue will be solved in Appium Desktop?

**Comment by dpgraham** [comment]
@simplezhang1990 when the next Appium version is released (1.13.0) we'll release an Appium Desktop version shortly after. The expected release for Appium 1.13 is before the end of April.
